### PR TITLE
Update security information in docs.

### PR DESCRIPTION
**Pull request by jof**
Update README.md with some updated security and trust information.

Additonally, update SECURITY.md to deprecate our email-based submission process with our current process via HackerOne.
This change also removes the now-unused PGP public key document.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -331,17 +331,20 @@ We offer a few different options for support. First of all, we try to provide cl
...
+information on our [trust page](https://trust.goteleport.com/).
 
 You can see the list of companies who use Teleport in production on the Teleport
 [product page](https://goteleport.com/case-study/).
 
-You can find the latest stable Teleport build on our [Releases](https://goteleport.com/teleport/download) page.
-
 ## Who Built Teleport?
 
 Teleport was created by [Gravitational Inc](https://goteleport.com). We have
```
Comment (reedloden) [inline README.md:350]:
```suggestion
Teleport was created by [Gravitational, Inc.](https://goteleport.com). We have
```

File: SECURITY.md [hunk 2]
```diff
@@ -7,9 +7,6 @@ The list of supported versions can be found
```
Comment (reedloden) [inline SECURITY.md:7]: I can't comment on the above line, but it should be:
```suggestion
[here](https://goteleport.com/download/).
```

File: README.md [hunk 3]
```diff
@@ -331,17 +331,20 @@ We offer a few different options for support. First of all, we try to provide cl
...
+Teleport has completed several security audits from nationally and
+internationally recognized technology security companies. 
+
+We publicize some of our audit results, security philosophy and related
+information on our [trust page](https://trust.goteleport.com/).
 
 You can see the list of companies who use Teleport in production on the Teleport
 [product page](https://goteleport.com/case-study/).
 
-You can find the latest stable Teleport build on our [Releases](https://goteleport.com/teleport/download) page.
```
Comment (reedloden) [inline README.md:343]: Was this meant to be removed? The URL needs to be updated at a minimum if it were to stay.

File: SECURITY.md (hunk 2 above)
Comment (jof) [reply SECURITY.md:7]: Addressed in a follow-up commit to update all the old download URLs.

File: README.md (hunk 3 above)
Comment (jof) [reply README.md:343]: I removed this from the "Is Teleport Secure and Production-Ready" section, as it was already explained further above in a more general section.

**Review by reedloden — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

File: SECURITY.md [hunk 4]
```diff
@@ -3,13 +3,10 @@
 ## Supported Versions
 
 The list of supported versions can be found
-[here](https://goteleport.com/teleport/download/).
+[here](https://goteleport.com/download/).
```
Comment (wadells) [inline SECURITY.md:6]: I'd link https://goteleport.com/docs/faq/#which-version-of-teleport-is-supported over the download page.

**Review by wadells — COMMENTED** [review]
Looks like I was a bit too slow. :smile: Nonetheless:

**Pull request merged**